### PR TITLE
Feature: Allow users to sign in with their OSF accounts via oauth [OC-62]

### PR DESCRIPTION
**Pull request by samanehsan**
Use django allauth to allow users to login using their OSF account

### CODE REVIEW

File: service/api/serializers.py [hunk 1]
```diff
@@ -5,25 +5,38 @@
...
 
-class UserSerializer(serializers.Serializer):
+class UserSerializer(serializers.ModelSerializer):
+
+    def __init__(self, *args, **kwargs):
+        super(UserSerializer, self).__init__(*args, **kwargs)
+        self.fields.update({
+            'token': serializers.SerializerMethodField()
+        })
+
```
Comment (birdbrained) [inline service/api/serializers.py]: Can this go with the other fields?

File: service/osf_oauth2_adapter/apps.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import os
+
+from django.apps import AppConfig
+
+
+class OsfOauth2AdapterConfig(AppConfig):
+    name = 'osf_oauth2_adapter'
+    # staging by default so people don't have to run OSF to use this.
+    osf_api_url = os.environ.get('OSF_API_URL', 'https://staging-api.osf.io').rstrip('/') + '/'
```
Comment (birdbrained) [inline service/osf_oauth2_adapter/apps.py]: You would only need rstrip() here if the env wasn't configured properly right?

File: service/osf_oauth2_adapter/provider.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+from .apps import OsfOauth2AdapterConfig
```
Comment (birdbrained) [inline service/osf_oauth2_adapter/provider.py]: Should we use absolute imports?

File: service/service/settings/base.py [hunk 4]
```diff
@@ -58,6 +64,8 @@
                 'django.template.context_processors.request',
                 'django.contrib.auth.context_processors.auth',
                 'django.contrib.messages.context_processors.messages',
+                # 'allauth.account.context_processors.account',
```
Comment (birdbrained) [inline service/service/settings/base.py]: dead code?

File: service/service/settings/base.py [hunk 5]
```diff
@@ -66,7 +74,41 @@
...
+LOGIN_REDIRECT_URL = os.environ.get('LOGIN_REDIRECT_URL', 'http://localhost:4200/')
+ACCOUNT_LOGOUT_REDIRECT_URL = os.environ.get('ACCOUNT_LOGOUT_REDIRECT_URL', '/api/')
+SOCIALACCOUNT_ADAPTER = 'osf_oauth2_adapter.views.OSFOAuth2Adapter'
+SOCIALACCOUNT_PROVIDERS = \
+    {'osf':
+        {
+            'METHOD': 'oauth2',
+            'SCOPE': ['osf.full_read'],
+            'AUTH_PARAMS': {'access_type': 'offline'},
+            # 'FIELDS': [
```
Comment (birdbrained) [inline service/service/settings/base.py]: dead?

File: service/osf_oauth2_adapter/apps.py (hunk 2 above)
Comment (samanehsan) [inline service/osf_oauth2_adapter/apps.py]: Yeah, it looks like if you forget the trailing slash in `'OSF_API_URL'` this adds it back

**Comment by samanehsan** [comment]
Done making changes @birdbrained! ☀️

**Comment by birdbrained** [comment]
sweet! I'm gonna merge it. 🍺

**Pull request merged**